crates/polars-python/src/lazyframe/visitor/expr_nodes.rs

```diff
@@ -465,6 +462,78 @@ impl PyRollingGroupOptions {
     }
 }
 
+#[pyclass(name = "DynamicGroupOptions")]
+pub struct PyDynamicGroupOptions {
+    inner: DynamicGroupOptions,
+}
+
+#[pymethods]
+impl PyDynamicGroupOptions {
+    #[getter]
+    fn index_column(&self, py: Python<'_>) -> PyResult<PyObject> {
+        Ok(self.inner.index_column.to_object(py))
+    }
+
+    #[getter]
+    fn every(&self, py: Python<'_>) -> PyResult<PyObject> {
+        Ok(Wrap(self.inner.every).into_py(py))
+    }
+
+    #[getter]
+    fn period(&self, py: Python<'_>) -> PyResult<PyObject> {
+        Ok(Wrap(self.inner.period).into_py(py))
+    }
+
+    #[getter]
+    fn offset(&self, py: Python<'_>) -> PyResult<PyObject> {
+        Ok(Wrap(self.inner.offset).into_py(py))
+    }
+
+    #[getter]
+    fn label(&self, py: Python<'_>) -> PyResult<PyObject> {
+        let result = match self.inner.label {
```

Comment:
We can tag `Label` with `#[derive(IntoStaticStr)]`

Reply:
question: I did not know about this derive macro previously, so there are a bunch of cases where I do this match dance. Is it worth tagging all those enums in polars with `IntoStaticStr` and then using this pattern here? This will be a breaking change (not a big deal) in the IR, since the names will change their case-pattern. e.g. `StartBy`, not `start_by`.

Reply:
We can even control the output with those macros. I believe `#[strum(serialize_all = "snake_case")]`, see: https://docs.rs/strum/latest/strum/additional_attributes/index.html

Reply:
Ah, even better. I'll try with that and see if anything breaks locally.